crates/iceberg/src/expr/predicate.rs

```diff
@@ -217,6 +249,10 @@ impl<T: Display + Debug> Display for SetExpression<T> {
 /// Unbound predicate expression before binding to a schema.
 #[derive(Debug, PartialEq)]
 pub enum Predicate {
+    /// AlwaysTrue predicate, for example, `TRUE`.
+    AlwaysTrue,
+    /// AlwaysFalse predicate, for example, `FALSE`.
+    AlwaysFalse,
```

Comment:
It's in here because it is needed in two places in `InclusiveProjection`:

- First, when we "unbind", transforming a `BoundPredicate` into an unbound `Predicate`: https://github.com/apache/iceberg-rust/pull/321/files#diff-419289c8023c272de50ee4352320ad217fa7c2802f7f633d26e55ece30426be6R41
- Second, when we build up a `Predicate` by "and"ing together a (possibly empty) list of `Predicate`s, starting with an `AlwaysTrue` to ensure that we return a valid `Predicate` even in the empty list case: https://github.com/apache/iceberg-rust/pull/321/files#diff-419289c8023c272de50ee4352320ad217fa7c2802f7f633d26e55ece30426be6R76

Reply:
Reason to me, thanks for the explaination.

Reply:
To add here, when we bind an `isNull` expression to a required field, then we know that that isn't possible :)